homeassistant/components/websocket_api/connection.py

```diff
@@ -37,6 +38,16 @@ def __init__(
         self.subscriptions: dict[Hashable, Callable[[], Any]] = {}
         self.last_id = 0
 
+        def revoke_callback() -> None:
+            self._token_revoked = True
```

Comment:
This should cause `WebSocketHandler._cancel` to be called directly. We shouldn't wait until the next message.